packages/flutter_tools/templates/create/lib/main.dart.tmpl

```diff
@@ -124,3 +101,45 @@ class _MyHomePageState extends State<MyHomePage> {
     );
   }
 }
+{{/withPluginHook}}
+{{#withPluginHook}}
+class _MyHomePageState extends State<MyHomePage> {
+  String _platformVersion = 'Unknown';
+
+  @override
+  initState() {
+    super.initState();
+    initPlatformState();
+  }
+
+  // Platform messages are asynchronous, so we initialize in an async method.
+  initPlatformState() async {
+    String platformVersion;
+    // Platform messages may fail, so we use a try/catch PlatformException.
+    try {
+      platformVersion = await {{pluginDartClass}}.platformVersion;
```

Comment:
It's unfortunate that there's no way to cancel the platformVersion call. Imagine if the call did a lot of computation, but the user triggered it then left the page that cares about the result. The work would still get done, but the result would then be thrown away.

Anyway that's an issue for another bug.